### PR TITLE
(GH-139) Clarification added regarding the DSC Local Configuration Manager (LCM)

### PR DESCRIPTION
**Pull request by david22swan**
In order for the Puppet dsc module to function correctly the value of the DSC Local Configuration Module (LCM) must be set to Push or Disabled.

### CODE REVIEW

File: src/internal/functions/Get-ReadmeContent.ps1 [hunk 1]
```diff
@@ -237,6 +237,16 @@ For specific information on troubleshooting a generated module, check the [troub
 Currently, because of the way Puppet caches files on agents, use of the legacy [``puppetlabs-dsc``]($LegacyDscForgePage) module is **not** compatible with this or any auto-generated DSC module.
 Inclusion of both will lead to pluginsync conflicts.
 
+### Configuring the LCM
+
+In order for the Puppet dsc module to function the DSC Local Configuration Manager (LCM) RefreshMode must be set to either Push or Disabled.
```
Comment (michaeltlombardi) [inline src/internal/functions/Get-ReadmeContent.ps1:242]:
```suggestion
In order for a Puppetized DSC module to function, the DSC Local Configuration Manager (LCM) ``RefreshMode`` must be set to either ``Push`` or ``Disabled``.
```

File: src/internal/functions/Get-ReadmeContent.ps1 [hunk 2]
```diff
@@ -237,6 +237,16 @@ For specific information on troubleshooting a generated module, check the [troub
...
+The default value for RefreshMode in WMF 5.0 and WMF 5.1 is Push — so if it has not been set to anything else then there is no action needed on your part.
```
Comment (michaeltlombardi) [inline src/internal/functions/Get-ReadmeContent.ps1:243]:
```suggestion
The default value for ``RefreshMode`` in WMF 5.0 and WMF 5.1 is ``Push`` — so if it has not been set to anything else then there is no action needed on your part.
```

File: src/internal/functions/Get-ReadmeContent.ps1 [hunk 3]
```diff
@@ -237,6 +237,16 @@ For specific information on troubleshooting a generated module, check the [troub
...
+However if the value of the LCM has been set to anything other than Push then the module will not function and so the value must either be changed back or disabled.
```
Comment (michaeltlombardi) [inline src/internal/functions/Get-ReadmeContent.ps1:244]:
```suggestion
However if the value of the LCM has been set to anything other than ``Push`` then the module will not function and so the value must either be changed back or disabled.
```

File: src/internal/functions/Get-ReadmeContent.ps1 [hunk 4]
```diff
@@ -237,6 +237,16 @@ For specific information on troubleshooting a generated module, check the [troub
...
+
+The Puppet dsc module uses the Invoke-DscResource cmdlet to invoke DSC Resources of the target machine.
```
Comment (michaeltlombardi) [inline src/internal/functions/Get-ReadmeContent.ps1:246]:
```suggestion
The Puppetized DSC modules use the ``Invoke-DscResource`` cmdlet to invoke DSC Resources of the target machine.
```

File: src/internal/functions/Get-ReadmeContent.ps1 [hunk 5]
```diff
@@ -237,6 +237,16 @@ For specific information on troubleshooting a generated module, check the [troub
...
+If the RefreshMode is set to Pull, DSC Resources will only run from a DSC Pull Server — in this setting DSC does not allow any DSC Resources to be run interactively on the host.
```
Comment (michaeltlombardi) [inline src/internal/functions/Get-ReadmeContent.ps1:247]:
```suggestion
If the ``RefreshMode`` is set to ``Pull``, DSC Resources will only run from a DSC Pull Server — in this setting DSC does not allow any DSC Resources to be run interactively on the host.
```

File: src/internal/functions/Get-ReadmeContent.ps1 [hunk 6]
```diff
@@ -237,6 +237,16 @@ For specific information on troubleshooting a generated module, check the [troub
...
+
+
```
Comment (michaeltlombardi) [inline src/internal/functions/Get-ReadmeContent.ps1:249]: Remove extra newline.

```suggestion
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - A check to `New-PuppetDscModule` which validates that PSRemoting is enabled and errors clearly if not in order to prevent unexpected execution failures later in the process ([#133](https://github.com/puppetlabs/Puppet.Dsc/issues/133))
 - Long file path support information to the requirements section of autogenerated READMEs ([#149](https://github.com/puppetlabs/Puppet.Dsc/issues/149))
 - Notes on installation limitations to generated README files to clarify minitar limitations for `r10k` and serverless Puppet users ([#150](https://github.com/puppetlabs/Puppet.Dsc/issues/150))
+- (GH-139) - Clarification added regarding the DSC Local Configuration Manager (LCM) ([#139](https://github.com/puppetlabs/Puppet.Dsc/issues/139))
```
Comment (michaeltlombardi) [inline CHANGELOG.md:16]: We can drop the ticket prepend at the start of the entry and reword to be explicit about the change.

```suggestion
- Clarification regarding configuring the DSC Local Configuration Manager (LCM) to work with Puppetized DSC modules ([#139](https://github.com/puppetlabs/Puppet.Dsc/issues/139))
```

**Review by michaeltlombardi — CHANGES_REQUESTED** [review]
Minor rewording suggestions and syntax changes.

**Comment by michaeltlombardi** [comment]
Tests are failing because of the em dashes in the added text - replacing those with `--` should fix the issue, I _think_.

**Comment by david22swan** [comment]
Updated

**Review by michaeltlombardi — APPROVED** [review]

**Pull request merged**